packages/flutter/lib/src/material/banner.dart

```diff
@@ -270,42 +282,50 @@ class _MaterialBannerState extends State<MaterialBanner> {
       ),
     );
 
+    final double elevation = widget.elevation ?? bannerTheme.elevation ?? 0.0;
     final Color backgroundColor = widget.backgroundColor
         ?? bannerTheme.backgroundColor
         ?? theme.colorScheme.surface;
     final TextStyle? textStyle = widget.contentTextStyle
         ?? bannerTheme.contentTextStyle
         ?? theme.textTheme.bodyText2;
 
-    Widget materialBanner = Container(
-      color: backgroundColor,
-      child: Column(
-        mainAxisSize: MainAxisSize.min,
-        children: <Widget>[
-          Padding(
-            padding: padding,
-            child: Row(
-              children: <Widget>[
-                if (widget.leading != null)
-                  Padding(
-                    padding: leadingPadding,
-                    child: widget.leading,
-                  ),
-                Expanded(
-                  child: DefaultTextStyle(
-                    style: textStyle!,
-                    child: widget.content,
-                  ),
+    Widget materialBanner = SafeArea(
+      top: true,
+      child: Container(
+        margin: EdgeInsets.only(bottom: elevation > 0 ? 10.0 : 0.0),
+        child: Material(
+          elevation: elevation,
+          color: backgroundColor,
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            children: <Widget>[
+              Padding(
+                padding: padding,
+                child: Row(
+                  children: <Widget>[
+                    if (widget.leading != null)
+                      Padding(
+                        padding: leadingPadding,
+                        child: widget.leading,
+                      ),
+                    Expanded(
+                      child: DefaultTextStyle(
+                        style: textStyle!,
+                        child: widget.content,
+                      ),
+                    ),
+                    if (isSingleRow)
+                      buttonBar,
+                  ],
                 ),
-                if (isSingleRow)
-                  buttonBar,
-              ],
-            ),
+              ),
+              if (!isSingleRow)
+                buttonBar,
+              const Divider(height: 0),
```

Comment:
I wonder if the Divider should be conditional on the elevation. Maybe if elevation>0 then we don't have a divider, since it seems weird to have both